lib/pavilion/dir_db.py

```diff
@@ -92,24 +97,73 @@ def create_id_dir(id_dir: Path) -> (int, Path):
 
         return next_id, next_id_path
 
-
 def default_filter(_: Path) -> bool:
     """Pass every path."""
 
     return True
 
+def is_hex(string: str) -> bool:
+    """Determines if passed string is valide hex or not.
+    :param string: A string to check.
+    :return bool: True if hex, False, if not hex.
+    """
+
+    try:
+        int(string, 16)
+    except ValueError as err:
+        return False
+
+    return True
 
 def select(id_dir: Path,
-           filter_func: Callable[[Path], bool] = default_filter) -> List[Path]:
+           filter_func: Callable[[Path], bool] = default_filter,
+           builds_dir: bool = False) -> List[Path]:
     """Return a list of all test paths in the given id_dir that pass the
     given filter function. The paths returned are guaranteed (within limits)
     to be an id directory, and only paths that pass the filter function
     are returned."""
 
     passed = []
     for path in id_dir.iterdir():
-        if path.name.isdigit() and path.is_dir():
-            if filter_func(path):
-                passed.append(path)
+        if not builds_dir:
+            if path.name.isdigit() and path.is_dir():
+                if filter_func(path):
+                    passed.append(path)
+        else:
+            if is_hex(path.name) and path.is_dir():
+                if filter_func(path):
+                    passed.append(path)
 
     return passed
+
+def delete(id_dir: Path,
+           filter_func: Callable[[Path], bool] = default_filter,
+           verbose: bool = False) -> int:
+    """Deletes a directory and it's entire contents.
+
+    :param id_dir: The directory to iterate through
+    :param verbose: True or False based on if you want verbose output. Default
+                    is false.
+    :return int count: The number of directories removed.
+    """
+
+    count = 0
+
+    lock_path = id_dir.with_suffix('.lock')
+    with lockfile.LockFile(lock_path):
+        for path in select(id_dir, filter_func):
+            try:
+                shutil.rmtree(path)
+            except OSError as err:
+                output.fprint("Could not remove {} {}: "
+                              "{}".format(id_dir.name, path, err),
+                              color=output.YELLOW)
+                continue
+            count += 1
+            if verbose:
```

Comment:
Again, return a list of messages along with the count, and let the caller deal with them. 

I acknowledge that this is awkward, but it makes usage outside of commands (like with unit-testing) a lot easier.